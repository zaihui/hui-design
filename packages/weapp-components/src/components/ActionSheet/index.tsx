import ActionSheet from './ActionSheet'
import Item from './Item/Item'
import type { HuiActionSheetProps } from './ActionSheet'
import type { HuiActionSheetItemProps } from './Item/Item'

type IActionSheet = typeof ActionSheet

interface IHuiActionSheet extends IActionSheet {
  Item: typeof Item
}

const HuiActionSheet = ActionSheet as IHuiActionSheet

HuiActionSheet.Item = Item

export type { HuiActionSheetProps }
export type { HuiActionSheetItemProps }
export default HuiActionSheet
