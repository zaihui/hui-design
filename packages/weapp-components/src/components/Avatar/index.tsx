import HuiAvatar from './Avatar'
import type { HuiAvatarProps } from './Avatar'

export type { HuiAvatarProps }
export default HuiAvatar
