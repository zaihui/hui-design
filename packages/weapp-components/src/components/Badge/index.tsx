import Badge from './Badge'
import type { HuiBadgeProps } from './Badge'

export type { HuiBadgeProps }
export default Badge
