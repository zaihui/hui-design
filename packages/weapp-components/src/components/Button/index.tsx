import Button from './Button'
import type { HuiButtonProps } from './Button'

export type { HuiButtonProps }
export default Button
