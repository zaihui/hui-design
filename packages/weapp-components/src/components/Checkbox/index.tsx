import React from 'react'
import InternalCheckbox from './Checkbox'
import HuiCheckboxGroup from './Group'
import type { HuiCheckboxProps, HuiCheckboxRef } from './Checkbox'
import type { HuiCheckboxGroupProps } from './Group'

interface CompundedComponent
  extends React.ForwardRefExoticComponent<HuiCheckboxProps> {
  Group: typeof HuiCheckboxGroup
}

const HuiCheckbox = InternalCheckbox as CompundedComponent
HuiCheckbox.Group = HuiCheckboxGroup

export type { HuiCheckboxGroupProps }
export type { HuiCheckboxProps, HuiCheckboxRef }
export default HuiCheckbox
