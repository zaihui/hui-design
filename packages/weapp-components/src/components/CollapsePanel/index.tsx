import type { CollapsePanelProps } from './CollapsePanel'
import CollapsePanel from './CollapsePanel'

export type { CollapsePanelProps }
export default CollapsePanel
