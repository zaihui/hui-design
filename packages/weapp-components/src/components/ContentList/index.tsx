import Item from './Item'
import type { HuiContentListItemProps } from './Item/index'

import ContentList from './ContentList'
import type { HuiContentListProps } from './ContentList'

interface CompundedComponent extends React.FC<HuiContentListProps> {
  Item: typeof Item
}

const HuiContentList = ContentList as CompundedComponent
HuiContentList.Item = Item

export type { HuiContentListProps }
export type { HuiContentListItemProps }
export default HuiContentList
