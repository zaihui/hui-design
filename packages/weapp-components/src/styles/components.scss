@import '../components/Toast/Toast.scss';
@import '../components/Dialog/Dialog.scss';
@import '../components/Input/Input.scss';
@import '../components/Button/Button.scss';
@import '../components/TextArea/TextArea.scss';
@import '../components/Divider/Divider.scss';
@import '../components/Loader/Loader.scss';
@import '../components/Switch/Switch.scss';
@import '../components/ActionSheet/ActionSheet.scss';
@import '../components/DefaultPage/DefaultPage.scss';
@import '../components/Checkbox/Checkbox.scss';
@import '../components/Radio/Radio.scss';
@import '../components/Tabs/Tabs.scss';
@import '../components/Sticky/Sticky.scss';
@import '../components/Tag/Tag.scss';
@import '../components/List/ListItem.scss';
@import '../components/Grid/Grid.scss';
@import '../components/Stepper/Stepper.scss';
@import '../components/Tooltip/Tooltip.scss';
@import '../components/SwiperIndicator/SwiperIndicator.scss';
@import '../components/Badge/Badge.scss';
@import '../components/NoticeBar/NoticeBar.scss';
@import '../components/Picker/Picker.scss';
@import '../components/Image/Image.scss';
@import '../components/Avatar/Avatar.scss';
@import '../components/Popup/Popup.scss';
@import '../components/Modal/Modal.scss';
@import '../components/Select/Select.scss';
@import '../components/Select/SideMenu/SideMenu.scss';
@import '../components/Select/Menu/Menu.scss';
@import '../components/Share/Share.scss';
@import '../components/Timer/Timer.scss';
@import '../components/Footer/Footer.scss';
@import '../components/Skeleton/Skeleton.scss';
@import '../components/NavBar/NavBar.scss';
@import '../components/EditorDisplay/EditorDisplay.scss';
@import '../components/Mask/Mask.scss';
@import '../components/FloatingActionButton/FloatingActionButton.scss';
@import '../components/Card/Card.scss';
@import '../components/Steppers/Steppers.scss';
@import '../components/Filter/Filter.scss';
@import '../components/Form/index.scss';
@import '../components/Filter/Menu/Menu.scss';
@import '../components/Filter/MenuItem/MenuItem.scss';
@import '../components/Filter/ActionFooter/ActionFooter.scss';
@import '../components/Filter/FiltersContent/FiltersContent.scss';
@import '../components/CollapsePanel/CollapsePanel.scss';
