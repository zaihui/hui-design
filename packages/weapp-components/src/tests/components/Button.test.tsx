import React from 'react'
import Taro from '@tarojs/taro'
import { Block } from '@tarojs/components'
import { findDOMNode } from 'nervjs'
import { renderToString } from 'nerv-server'
import { Simulate, renderIntoDocument } from 'nerv-test-utils'

import HuiButton from '@/components/Button/Button'

Taro.initPxTransform({ designWidth: 375, deviceRatio: {} })

describe('huiButton Snap', () => {
  it('renders HuiButton', () => {
    expect.hasAssertions()
    const component = renderToString(<HuiButton>按钮文字</HuiButton>)
    expect(component).toMatchSnapshot()
  })

  it('renders HuiButton - type', () => {
    expect.hasAssertions()
    const component = renderToString(<HuiButton type='primary'>主操作按钮</HuiButton>)
    expect(component).toMatchSnapshot()
  })

  it('renders HuiButton - width', () => {
    expect.hasAssertions()
    const component = renderToString(<HuiButton width={188}>自定义宽度</HuiButton>)
    expect(component).toMatchSnapshot()
  })

  it('renders HuiButton - block', () => {
    expect.hasAssertions()
    const component = renderToString(<HuiButton block>按钮文字</HuiButton>)
    expect(component).toMatchSnapshot()
  })

  it('renders HuiButton - size', () => {
    expect.hasAssertions()
    const component = renderToString(
      <Block>
        <HuiButton size='small'>按钮文字</HuiButton>
        <HuiButton size='medium'>按钮文字</HuiButton>
        <HuiButton size='large'>按钮文字</HuiButton>
      </Block>,
    )
    expect(component).toMatchSnapshot()
  })

  it('renders HuiButton - color', () => {
    expect.hasAssertions()
    const component = renderToString(
      <Block>
        <HuiButton size='small' color='#ed3737'>
          按钮文字
        </HuiButton>
        <HuiButton size='small' color='#ff6c00'>
          按钮文字
        </HuiButton>
      </Block>,
    )
    expect(component).toMatchSnapshot()
  })

  it('renders HuiButton - disabled', () => {
    expect.hasAssertions()
    const component = renderToString(<HuiButton disabled>按钮文字</HuiButton>)
    expect(component).toMatchSnapshot()
  })

  it('renders HuiButton - icon', () => {
    expect.hasAssertions()
    const component = renderToString(
      <Block>
        <HuiButton prefixIcon='003-right'>按钮文字</HuiButton>
        <HuiButton suffixIcon='003-right'>按钮文字</HuiButton>
      </Block>,
    )
    expect(component).toMatchSnapshot()
  })

  it('renders HuiButton - extra', () => {
    expect.hasAssertions()
    const component = renderToString(
      <HuiButton extra='辅助文字' size='large'>
        按钮文字
      </HuiButton>,
    )
    expect(component).toMatchSnapshot()
  })

  it('renders HuiButton - extra with size m', () => {
    expect.hasAssertions()
    const component = renderToString(<HuiButton extra='辅助文字'>按钮文字</HuiButton>)
    expect(component).toMatchSnapshot()
  })
})

describe('huiButton Event', () => {
  it('huiButton onClick', () => {
    expect.hasAssertions()
    const onClick = jest.fn()
    const component = renderIntoDocument(<HuiButton onClick={onClick}>按钮文字</HuiButton>)
    const componentDom = findDOMNode(component, 'hui-button')
    Simulate.click(componentDom)
    expect(onClick).toHaveBeenCalled()
  })

  it('huiButton onClick - disabled', () => {
    expect.hasAssertions()
    const onClick = jest.fn()
    const component = renderIntoDocument(
      <HuiButton disabled onClick={onClick}>
        按钮文字
      </HuiButton>,
    )
    const componentDom = findDOMNode(component, 'hui-button')
    Simulate.click(componentDom)
    expect(onClick).not.toHaveBeenCalled()
  })
})
