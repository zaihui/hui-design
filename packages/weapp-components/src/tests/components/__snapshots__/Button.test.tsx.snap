// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`huiButton Snap renders HuiButton - block 1`] = `"<button style=\\"background:;border-color:;\\" class=\\"hui-button medium primary radius-default\\">undefined<div>undefined<div class=\\"content-wrapper\\"><div class=\\"\\">按钮文字</div>undefined</div>undefined</div>undefined</button>"`;

exports[`huiButton Snap renders HuiButton - color 1`] = `"<div><button style=\\"background:#ed3737;border-color:#ed3737;\\" class=\\"hui-button small primary radius-default inline\\">undefined<div>undefined<div class=\\"content-wrapper\\"><div class=\\"\\">按钮文字</div>undefined</div>undefined</div>undefined</button><button style=\\"background:#ff6c00;border-color:#ff6c00;\\" class=\\"hui-button small primary radius-default inline\\">undefined<div>undefined<div class=\\"content-wrapper\\"><div class=\\"\\">按钮文字</div>undefined</div>undefined</div>undefined</button></div>"`;

exports[`huiButton Snap renders HuiButton - disabled 1`] = `"<button style=\\"background:;border-color:;\\" disabled class=\\"hui-button medium primary radius-default disabled inline\\">undefined<div>undefined<div class=\\"content-wrapper\\"><div class=\\"\\">按钮文字</div>undefined</div>undefined</div>undefined</button>"`;

exports[`huiButton Snap renders HuiButton - extra 1`] = `"<button style=\\"background:;border-color:;\\" class=\\"hui-button large primary radius-default inline\\">undefined<div>undefined<div class=\\"content-wrapper\\"><div class=\\"\\">按钮文字</div><div class=\\"extra\\">辅助文字</div></div>undefined</div>undefined</button>"`;

exports[`huiButton Snap renders HuiButton - extra with size m 1`] = `"<button style=\\"background:;border-color:;\\" class=\\"hui-button medium primary radius-default inline\\">undefined<div>undefined<div class=\\"content-wrapper\\"><div class=\\"\\">按钮文字</div><div class=\\"extra\\">辅助文字</div></div>undefined</div>undefined</button>"`;

exports[`huiButton Snap renders HuiButton - icon 1`] = `"<div><button style=\\"background:;border-color:;\\" class=\\"hui-button medium primary radius-default inline\\">undefined<div><div style=\\"color:inherit;font-size:0.5974rem;margin-right:0.384rem;\\" class=\\"hui-icons hui-icon-001-success\\"></div><div class=\\"content-wrapper\\"><div class=\\"\\">按钮文字</div>undefined</div>undefined</div>undefined</button><button style=\\"background:;border-color:;\\" class=\\"hui-button medium primary radius-default inline\\">undefined<div>undefined<div class=\\"content-wrapper\\"><div class=\\"\\">按钮文字</div>undefined</div><div style=\\"color:inherit;font-size:0.5974rem;margin-left:0.384rem;\\" class=\\"hui-icons hui-icon-001-success\\"></div></div>undefined</button></div>"`;

exports[`huiButton Snap renders HuiButton - size 1`] = `"<div><button style=\\"background:;border-color:;\\" class=\\"hui-button small primary radius-default inline\\">undefined<div>undefined<div class=\\"content-wrapper\\"><div class=\\"\\">按钮文字</div>undefined</div>undefined</div>undefined</button><button style=\\"background:;border-color:;\\" class=\\"hui-button medium primary radius-default inline\\">undefined<div>undefined<div class=\\"content-wrapper\\"><div class=\\"\\">按钮文字</div>undefined</div>undefined</div>undefined</button><button style=\\"background:;border-color:;\\" class=\\"hui-button large primary radius-default inline\\">undefined<div>undefined<div class=\\"content-wrapper\\"><div class=\\"\\">按钮文字</div>undefined</div>undefined</div>undefined</button></div>"`;

exports[`huiButton Snap renders HuiButton - type 1`] = `"<button style=\\"background:;border-color:;\\" class=\\"hui-button medium primary radius-default inline\\">undefined<div>undefined<div class=\\"content-wrapper\\"><div class=\\"\\">主操作按钮</div>undefined</div>undefined</div>undefined</button>"`;

exports[`huiButton Snap renders HuiButton - width 1`] = `"<button style=\\"width:8.0214rem;background:;border-color:;\\" class=\\"hui-button medium primary radius-default inline\\">undefined<div>undefined<div class=\\"content-wrapper\\"><div class=\\"\\">自定义宽度</div>undefined</div>undefined</div>undefined</button>"`;

exports[`huiButton Snap renders HuiButton 1`] = `"<button style=\\"background:;border-color:;\\" class=\\"hui-button medium primary radius-default inline\\">undefined<div>undefined<div class=\\"content-wrapper\\"><div class=\\"\\">按钮文字</div>undefined</div>undefined</div>undefined</button>"`;
